File: edukate-backend/src/main/java/io/github/sanyavertolet/edukate/backend/services/ShareCodeGenerator.java
package io.github.sanyavertolet.edukate.backend.services;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

import java.security.SecureRandom;

@Component
@RequiredArgsConstructor
public class ShareCodeGenerator {
    private static final String ALLOWED_CHARACTERS = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_";
    private static final int CODE_LENGTH = 10;
    private final SecureRandom random = new SecureRandom();

    public String generateShareCode() {
        StringBuilder sb = new StringBuilder(CODE_LENGTH);
        for (int i = 0; i < CODE_LENGTH; i++) {
            sb.append(ALLOWED_CHARACTERS.charAt(random.nextInt(ALLOWED_CHARACTERS.length())));
        }
        return sb.toString();
    }
}

File: edukate-backend/src/main/kotlin/io/github/sanyavertolet/edukate/backend/services/ShareCodeGenerator.kt
package io.github.sanyavertolet.edukate.backend.services

import java.security.SecureRandom
import org.springframework.stereotype.Component

@Component
class ShareCodeGenerator {
    private val random = SecureRandom()

    fun generateShareCode(): String =
        (1..CODE_LENGTH).map { ALLOWED_CHARACTERS[random.nextInt(ALLOWED_CHARACTERS.length)] }.joinToString("")

    companion object {
        private const val ALLOWED_CHARACTERS = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789"
        private const val CODE_LENGTH = 10
    }
}
